selfhost/types.jakt

```diff
@@ -638,6 +638,50 @@ class CheckedFunction {
     public is_virtual: bool
     public is_override: bool
 
+    public function signature_matches(this, anon other: CheckedFunction) throws -> bool {
```

Comment:
Would `Typechecker::signatures_match()` work for this?

Reply:
As far as I understand the implementation: no since those serve different purposes
- `CheckedFunction::signature_matches ` checks if all arguments are the same i.e. `f<T>(x: T)` is **not the same** as `f(x: String)`
- `Typechecker::signatures_match` checks if all arguments are compatible `f<T>(x: T)` is **compatible with** `f(x: String)`

but since this function is used to check if a function with the same signature is already defined we want the former behaviour here, otherwise we would not allow more specific overrides of generic functions.